test/libsolidity/syntaxTests/storageLayoutSpecifier/constant_from_base_contract.sol

```
@@ -4,4 +4,4 @@ contract A {
 
 contract C is A layout at A.x { }
 // ----
-// TypeError 6396: (68-71): The base slot of the storage layout must evaluate to a rational number.
+// TypeError 1505: (68-71): The base slot expression cannot be evaluated during compilation.
```

This one is weird. Why are we getting this and not something like:
```
Error: Member "x" not found or not visible after argument-dependent lookup in type(contract A).
```

Is `ConstantEvaluator` silencing this error and just returning `nullopt`?

---

Yes, `ConstantEvaluator` can't handle member access. At the moment it can only process, literals, variables with no member acess from modules or other contracts and most of binary/unary operations.

---

Interesting. So apparently it's not just layouts, but `ConstantEvaluator` does not support such constants in general. It does not seem hard to add that though. Looking at `ConstantEvaluator`, it may be as simple as defining `endVisit(MemberAccess)` similar to `endVisit(Identifier)` it already has. I think we should do it: #16055.

This should not block this PR though. It's fine to leave it unsupported. You could add an explicit check for `MemberAccess` to `PostTypeContractLevelChecker` to issue a better message, but it will only catch cases where it's the top-level expression, so it's a half-measure.